src/engine/enginepregain.cpp

```diff
@@ -51,36 +53,46 @@ EnginePregain::EnginePregain(QString group)
 EnginePregain::~EnginePregain()
 {
     delete m_pPotmeterPregain;
-    delete m_pControlReplayGain;
+    delete m_pCOReplayGain;
     delete m_pTotalGain;
 
     delete s_pEnableReplayGain;
     s_pEnableReplayGain = NULL;
     delete s_pReplayGainBoost;
     s_pReplayGainBoost = NULL;
+    delete s_pDefaultBoost;
+    s_pDefaultBoost = NULL;
 }
 
 void EnginePregain::process(CSAMPLE* pInOut, const int iBufferSize) {
-
-    float fEnableReplayGain = s_pEnableReplayGain->get();
-    const float fReplayGainBoost = s_pReplayGainBoost->get();
-    float fGain = m_pPotmeterPregain->get();
-    float fReplayGain = m_pControlReplayGain->get();
-    float fReplayGainCorrection = 1;
-    float fPassing = m_pPassthroughEnabled->get();
-
-    // Override replaygain value if passing through
-    if (fPassing > 0) {
-        fReplayGain = 1.0;
-    } else if (fReplayGain * fEnableReplayGain != 0) {
-        // Here is the point, when ReplayGain Analyser takes its action, suggested gain changes from 0 to a nonzero value
+    const float fReplayGain = m_pCOReplayGain->get();
+    float fReplayGainCorrection;
+    if (m_pPassthroughEnabled->toBool()) {
+        // Override replaygain value if passing through
+        // TODO(XXX): consider a good default.
+        // Do we expect an replaygain leveled input or
+        // Normalized to 1 input?
+        fReplayGainCorrection = 1; // We expect a replaygain leveled input
+    } else if (!s_pEnableReplayGain->toBool() || fReplayGain == 0) {
```

Comment:
I understand that this is correct for your type of library. But as you are now happy with 0 dB DefaultBoost without ReplayGain, I need a -6 dB DefaultBoost for my library (what we effectively head in Mixxx 1.11). 
Since one Boost value is allways applied and bith are preference option now, you can tweak them at any time. We both should be happy with the current DefaultBoost that is always applied. Of cause I prefer the -6 dB and since that is where Mixxx is coming from and it fits to my library, I would like to keep it as default. 
But if you think the majority of users has non compressed Club tracks, we can change the default to 0 dB. 

But for my feeling, the defaults will matters most to casual users. Since ReplayGain is enabled anyway for them, a Default Boost < 0 dB will work best, because even for the majority of club tracks the ReplayGain is lower that the original  gain. I also think that the majority of casual users will have a library with tracks in "Radio edit" mode, leveled at 1.